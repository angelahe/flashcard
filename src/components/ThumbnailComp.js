import React from 'react';
import '../styles/flashcard.css';
import PropTypes from 'prop-types';

const ThumbnailComp = (props) => {
  const {
    imageUrl,
    imageId,
    onImageSelect,
  } = props;

  function handleThumbnailClick() {
    onImageSelect(imageUrl, imageId);
  }
  return (
    <div onClick={handleThumbnailClick} onKeyPress={handleThumbnailClick} role="button" tabIndex="0">
      <img className="thumbnailImg" src={imageUrl} alt={imageId} />
    </div>
  );
};

ThumbnailComp.defaultProps = {
  onImageSelect: () => { },
};
ThumbnailComp.propTypes = {
  imageUrl: PropTypes.string.isRequired,
  imageId: PropTypes.string.isRequired,
  onImageSelect: PropTypes.func,
};
export default ThumbnailComp;
