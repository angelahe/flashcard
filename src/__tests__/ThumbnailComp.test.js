import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import ThumbnailComp from '../components/ThumbnailComp';

const imageUrl = 'https://d30y9cdsu7xlg0.cloudfront.net/png/75112-200.png';
const imageId = '75112';
it('renders without crashing', () => {
  shallow(<ThumbnailComp imageUrl={imageUrl} imageId={imageId} />);
});

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<ThumbnailComp imageUrl={imageUrl} imageId={imageId} />, div);
  ReactDOM.unmountComponentAtNode(div);
});

test('snapshot of UI renders consistently', () => {
  const tree = renderer.create(<ThumbnailComp imageUrl={imageUrl} imageId={imageId} />).toJSON();
  expect(tree).toMatchSnapshot();
});
