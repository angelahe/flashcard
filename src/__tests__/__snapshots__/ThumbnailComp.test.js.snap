// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot of UI renders consistently 1`] = `
<div
  onClick={[Function]}
  onKeyPress={[Function]}
  role="button"
  tabIndex="0"
>
  <img
    alt="75112"
    className="thumbnailImg"
    src="https://d30y9cdsu7xlg0.cloudfront.net/png/75112-200.png"
  />
</div>
`;
